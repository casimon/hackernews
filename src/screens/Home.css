.Home {
  text-align: center;
}

.Home-header {
  background-color: #282c34;
  padding: 20px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.Home-body {
  padding: 20px;
}
