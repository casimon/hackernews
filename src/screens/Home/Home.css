.Page-body section {
  padding: 20px;
}
