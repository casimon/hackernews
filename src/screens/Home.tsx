import React from 'react'
import { useSelector } from 'react-redux'

import './Home.css'
import Card from '../components/Card/Card'
import { useFetching } from '../hooks/useFetching'
import { getTopStories } from '../reducers'

interface RootState {
  topStories: any
}

interface Story {
  by: string
  descendants: number
  id: number
  kids: Array<number>
  score: number
  time: number
  title: string
  type: string
  url: string
}

function Home() {
  useFetching(getTopStories)
  const topStories = useSelector((state: RootState) => state.topStories)
  console.log(topStories)
  return (
    <div className='Home'>
      <header className='Home-header'>
        <h1>Hacker News</h1>
      </header>
      <section className='Home-body'>
        <h2>Top Stories</h2>
        {topStories.map((item: Story, index: number) => (
          <Card title={item.title} url={item.url} key={index} />
        ))}
        <h2>Top Comments</h2>
      </section>
    </div>
  )
}

export default Home
