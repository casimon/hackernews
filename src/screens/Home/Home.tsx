import React from 'react'
import { useSelector } from 'react-redux'

import './Home.css'
import Title from '../../components/Title/Title'
import Header from '../../components/Header/Header'

import { useFetching } from '../../hooks/useFetching'
import { getTopStories } from '../../reducers'

interface Story {
  by: string
  descendants: number
  id: number
  kids: Array<number>
  score: number
  time: number
  title: string
  type: string
  url: string
}

interface TopStories {
  topStories: Array<Story>
}

function Home() {
  useFetching(getTopStories)
  const topStories = useSelector((state: TopStories) => state.topStories)
  return (
    <div className='Page'>
      <Header />
      <article className='Page-body'>
        <section>
          <h2>Top Stories</h2>
          {topStories.length > 0 ? (
            topStories.map((item: Story, index: number) => (
              <Title
                title={item.title}
                url={item.url}
                score={item.score}
                by={item.by}
                time={item.time}
                key={index}
              />
            ))
          ) : (
            <p>Loading...</p>
          )}
        </section>
      </article>
    </div>
  )
}

export default Home
