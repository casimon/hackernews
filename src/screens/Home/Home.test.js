import React from 'react'

import { Provider } from 'react-redux'
import { render } from '@testing-library/react'
import configureStore from 'redux-mock-store'
import Home from './Home'

const mockStore = configureStore([])

jest.mock('../../hooks/useFetching')

const init = (store) =>
  render(
    <Provider store={store}>
      <Home />
    </Provider>
  )

describe('My connected React-Redux component', () => {
  test('should render home page with default state from Redux store', () => {
    const store = mockStore({
      topStories: [],
      errors: [],
    })

    const { container } = init(store)
    expect(container).toMatchSnapshot()
  })

  test('should render home page with top stories', () => {
    const store = mockStore({
      topStories: [
        {
          by: 'test1',
          descendants: 1,
          id: 1,
          kids: [1],
          score: 1,
          time: 1,
          title: 'Test 1',
          type: 'story',
          url: 'http://www.test.com',
        },
      ],
      errors: [],
    })
    const { container } = init(store)
    expect(container).toMatchSnapshot()
  })
})
