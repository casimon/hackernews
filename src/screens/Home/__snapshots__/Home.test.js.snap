// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My connected React-Redux component should render home page with default state from Redux store 1`] = `
<div>
  <div
    class="Page"
  >
    <header
      class="Header"
    >
      <img
        alt="Logo"
        src="Y_Combinator_logo.svg"
      />
      <h1>
        Hacker News
      </h1>
      <div
        class="Header-nav"
      >
        <a
          href="/"
        >
          Home
        </a>
        <a
          href="/comments"
        >
          Comments
        </a>
      </div>
    </header>
    <article
      class="Page-body"
    >
      <section>
        <h2>
          Top Stories
        </h2>
        <p>
          Loading...
        </p>
      </section>
    </article>
  </div>
</div>
`;

exports[`My connected React-Redux component should render home page with top stories 1`] = `
<div>
  <div
    class="Page"
  >
    <header
      class="Header"
    >
      <img
        alt="Logo"
        src="Y_Combinator_logo.svg"
      />
      <h1>
        Hacker News
      </h1>
      <div
        class="Header-nav"
      >
        <a
          href="/"
        >
          Home
        </a>
        <a
          href="/comments"
        >
          Comments
        </a>
      </div>
    </header>
    <article
      class="Page-body"
    >
      <section>
        <h2>
          Top Stories
        </h2>
        <div
          class="Title"
        >
          <a
            class="Title-link"
            href="http://www.test.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            Test 1
          </a>
          <br />
          <span>
            Score - 1, By test1
          </span>
        </div>
      </section>
    </article>
  </div>
</div>
`;
